app/src/main/java/com/koalatea/thehollidayinn/softwareengineeringdaily/data/mapper/PostItemMapper.java

```diff
@@ -0,0 +1,32 @@
+package com.koalatea.thehollidayinn.softwareengineeringdaily.data.mapper;
+
+import com.koalatea.thehollidayinn.softwareengineeringdaily.data.models.PostItem;
+import com.koalatea.thehollidayinn.softwareengineeringdaily.network.response.PostResponse;
+
+/**
+ * Created by Kurian on 26-Sep-17.
+ */
+public class PostItemMapper extends BaseDataMapper<PostResponse, PostItem> {
```

Comment:
What is the benefit to manually mapping?

Reply:
Right now, just a way of isolating changes to the API from the in-app model. Potentially could also be used between local db models and presentation models. If there's a more efficient way of doing it while achieving the same result?

Reply:
You can add annotations for Gson to the properties you want to be parsed from the api result and ignore other properties. http://www.javacreed.com/gson-annotations-example/

Reply:
The concern I have is mixing domain/app data with API data by using the same container class for both. It can get quite messy and lead to multiple responsibilities. Keeping them as separate POJOs and manually mapping is more verbose but, makes for clearer readability in terms of their responsibilities.